tensorflow/stream_executor/rocm/rocm_dnn.cc

```diff
@@ -0,0 +1,4547 @@
+  // they will work with.
+  options.stack_size = 2 * 1024 * 1024;
+  options.guard_size = 2 * 1024 * 1024;
+  miopen_threadpool_ = new port::ThreadPool(port::Env::Default(), options,
+                                            "miopen_threadpool", 1);
+  CHECK(miopen_threadpool_);
+  return miopen_threadpool_;
+}
+
+static mutex miopen_threadpool_mu(LINKER_INITIALIZED);
+static port::ThreadPool* GetROCmThreadpool() {
+  mutex_lock lock(miopen_threadpool_mu);
+  static port::ThreadPool* miopen_threadpool = InitMIOpenThreadpool();
+  return miopen_threadpool;
+}
+
+#ifdef PLATFORM_GOOGLE
+
+#define STREAM_EXECUTOR_MIOPEN_WRAP(__name)      \
+  struct WrapperShim__##__name {                 \
+    template <typename... Args>                  \
+    miopenStatus_t operator()(Args... args) {    \
+      miopenStatus_t retval = ::__name(args...); \
+      return retval;                             \
+    }                                            \
+  } __name;
+
+#else
+
+#define STREAM_EXECUTOR_MIOPEN_WRAP(__name)                               \
+  struct DynLoadShim__##__name {                                          \
+    static const char* kName;                                             \
+    using FuncPtrT = std::add_pointer<decltype(::__name)>::type;          \
+    static void* GetDsoHandle() {                                         \
+      auto s = internal::CachedDsoLoader::GetMiopenDsoHandle();           \
+      return s.ValueOrDie();                                              \
+    }                                                                     \
+    static FuncPtrT LoadOrDie() {                                         \
+      void* f;                                                            \
+      auto s = port::Env::Default()->GetSymbolFromLibrary(GetDsoHandle(), \
+                                                          kName, &f);     \
+      CHECK(s.ok()) << "could not find " << kName                         \
+                    << " in miopen DSO; dlerror: " << s.error_message();  \
+      return reinterpret_cast<FuncPtrT>(f);                               \
+    }                                                                     \
+    static FuncPtrT DynLoad() {                                           \
+      static FuncPtrT f = LoadOrDie();                                    \
+      return f;                                                           \
+    }                                                                     \
+    template <typename... Args>                                           \
+    miopenStatus_t operator()(Args... args) {                             \
+      return DynLoad()(args...);                                          \
+    }                                                                     \
+  } __name;                                                               \
+  const char* DynLoadShim__##__name::kName = #__name;
+
+#endif
+
+// clang-format off
```

Comment:
What you are saying is correct, but it would make this macro inconsistent with all the other STREAM_EXECUTOR_*_WRAP macros in the rocm*.cc and cuda*.cc files. I would prefer that this be kept as is. Is that okay?

Reply:
> Is that okay?
Yes, absolutely fine.